optuna/integration/cma.py

```diff
@@ -422,7 +422,8 @@ def _to_optuna_params(search_space, param_name, cma_param_value):
             # v may slightly exceed range due to round-off errors.
             return float(min(max(v, dist.low), dist.high))
         if isinstance(dist, IntUniformDistribution):
-            return int(numpy.round(cma_param_value))
+            v = numpy.round(cma_param_value / dist.q) * dist.q + dist.low
+            return int(min(max(v, dist.low), dist.high))
```

Comment:
```suggestion
            r = numpy.round((cma_param_value - dist.low) / dist.q)
            v = r * dist.q + dist.low
            return v
```

* In the CmaEsSampler, `cma.tell()` is called without shifting IntUniformDistributions. So it's ok to just round `cma_param_value`.
    * https://github.com/optuna/optuna/blob/5db3bbfb8def796ac8fd2f2e81e2ef105540d003/optuna/integration/cma.py#L325-L347
* `pycma` provides BoundaryConstraints option. It ensures `cma_param_value` is in the range. It doesn't require to call `min(max(v, dist.low), dist.high)`
    * https://github.com/optuna/optuna/blob/5db3bbfb8def796ac8fd2f2e81e2ef105540d003/optuna/integration/cma.py#L310-L313

Reply:
Thanks a lot!